### PR TITLE
ambient: split ztunnel daemonset to its own chart

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

part of https://github.com/istio/istio/issues/41281

### CODE REVIEW

File: manifests/charts/ztunnel/.helmignore [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
```
Comment (howardjohn) [inline manifests/charts/ztunnel/.helmignore:1]: how come we don't need this in any other chart?

File: manifests/charts/ztunnel/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+name: ztunnel
+# This version is never actually shipped. istio/release-builder will replace it at build-time
+# with the appropriate version
+version: 0.1.0
```
Comment (howardjohn) [inline manifests/charts/ztunnel/Chart.yaml:5]: nit: I would replace with `version: 1.0.0` to match others

**Review by howardjohn — APPROVED** [review]

File: manifests/charts/ztunnel/.helmignore (hunk 1 above)
Comment (zirain) [reply manifests/charts/ztunnel/.helmignore:1]: it's auto generated by `helm create`
IMHO, this's no bad, but I'm ok to remove these

**Pull request merged**